src/simulation.jl

```diff
@@ -143,45 +165,85 @@ end
 
 is_form(d::SummationDecapode, var_name::Symbol) = is_form(d, first(incident(d, var_name, :name)))
 
+function getgeneric_type(type::Symbol)
+  if (type == :Form0 || type == :Form1 || type == :Form2 ||
+    type == :DualForm0 || type == :DualForm1 || type == :DualForm2)
+    return :Form
+  end
+  return type
+end
+
 is_literal(d::SummationDecapode, var_id::Int) = (d[var_id, :type] == :Literal)
 is_literal(d::SummationDecapode, var_name::Symbol) = is_literal(d, first(incident(d, var_name, :name)))
 
 is_infer(d::SummationDecapode, var_id::Int) = (d[var_id, :type] == :infer)
 is_infer(d::SummationDecapode, var_name::Symbol) = is_infer(d, first(incident(d, var_name, :name)))
 
-add_stub(stub_name::Symbol, var_name::Symbol) = return Symbol("$(stub_name)_$(var_name)")
+struct InvalidStubException <: Exception
+  stub::Symbol
+end
+
+Base.showerror(io::IO, e::InvalidStubException) = print(io, "Stub \"$(e.stub)\" is invalid")
+
+function add_stub(stub_name::Symbol, var_name::Symbol)
+  # No empty stubs
```

Comment:
It seems nice to have a "unit" empty stub. Is there a place in which sometimes empty stubs are returned because a lack of data somewhere, etc.? Or is this useful for the purposes of avoiding name collisions?

Reply:
I guess there's nothing technically wrong with an empty stub but in principle all stubs should have some description of what they're meant for.